diff --git shellinabox/shell_in_a_box.jspp.old shellinabox/shell_in_a_box.jspp
index 976c7e5..abd0ab3 100644
--- shellinabox/shell_in_a_box.jspp.old
+++ shellinabox/shell_in_a_box.jspp
@@ -474,11 +474,13 @@ ShellInABox.prototype.about = function() {
         (typeof serverSupportsSSL != 'undefined' && serverSupportsSSL ?
          "\n\n" +
          "This product includes software developed by the OpenSSL Project for\n" +
-         "use in the OpenSSL Toolkit. (http://www.openssl.org/)" +
+         "use in the OpenSSL Toolkit (http://www.openssl.org/)." +
          "\n\n" +
          "This product includes cryptographic software written by Eric Young\n" +
-         "(eay@cryptsoft.com)" :
-         ""));
+         "(eay@cryptsoft.com)." :
+         "") + 
+         "\n\n" +
+         "Ported and compiled into a package for use on Synology DSM by Philip Hughes (http://hughesbox.co.uk/).");
 };
 
 /* vim: set filetype=javascript : */
